### PR TITLE
Fix leaks in various GLKit classes by enabling ARC

### PR DESCRIPTION
**Pull request by ehren**
Noticed this earlier (should've filed an issue!). This is likely partially responsible for #2624 though may not completely solve the issue.

I'm just completing the build / verifying, will add a comment once ready for review.

### CODE REVIEW

File: Frameworks/GLKit/GLKEffect.mm [hunk 1]
```diff
@@ -528,7 +529,6 @@ - (BOOL)prepareShaders {
         self.shader = [[GLKShaderCache get] addShaderNamed:self.shaderName source:p];
         if (self.shader == nil) {
             NSTraceError(TAG, @"There was a problem generating a shader for material %@", self.shaderName);
-            [p release];
```
Comment (ehren) [inline Frameworks/GLKit/GLKEffect.mm]: ```addShaderNamed:source:``` doesn't retain ```p``` so this is missing a release in the non-error case were ARC to be left disabled

**Comment by DHowett-MSFT** [comment]
It scares me that turning on ARC was so easy because _nobody used `release` in GLKit` :wink:

**Comment by ehren** [comment]
no issues running WOCCatalog

**Comment by DHowett-MSFT** [comment]
Thank you, @ehren!

**Pull request merged**